api/src/main/java/org/openmrs/module/pihcore/htmlformentry/action/AddPatientToQueueAction.java

```diff
@@ -38,72 +36,68 @@ public class AddPatientToQueueAction implements CustomFormSubmissionAction {
     @Override
     public void applyAction(FormEntrySession formEntrySession) {
 
-        Concept addToQueueConcept = Context.getConceptService().getConceptByMapping(ADD_TO_QUEUE_CONCEPT_CODE_PIH, "PIH");
+        if (formEntrySession.getContext().getMode().equals(FormEntryContext.Mode.ENTER)) {
+            // we execute this post submission action only in the ENTER mode
+            Concept addToQueueConcept = Context.getConceptService().getConceptByMapping(ADD_TO_QUEUE_CONCEPT_CODE_PIH, "PIH");
 
-        if (addToQueueConcept == null) {
-            throw new RuntimeException("AddPatientToQueueAction not executing as Concept PIH: " + ADD_TO_QUEUE_CONCEPT_CODE_PIH + " cannot be found");
-        }
-        Patient patient = formEntrySession.getPatient();
-        Encounter encounter = formEntrySession.getEncounter();
-        Visit visit = encounter.getVisit();
-        if ( visit != null && visit.getStopDatetime() != null) {
-            // this post submit action do not apply to retrospective entries, only for active point of care visits
-            log.warn("AddPatientToQueueAction not executing because this is not an active visit");
-            // we should not throw an error, because the user should be able to edit other obs on this check-in encounter
-            return;
-        }
```

Comment:
Why are you removing the above block @cioan ?  It's still possible to have a non-active visit in Enter mode.  Do we not want this check any more?

Reply:
If the visit is not active then the Refer to triage question is not included in the form:

https://github.com/PIH/openmrs-config-pihsl/blob/master/configuration/pih/htmlforms/checkinMaternal.xml#L259

But, I could add back this code if you think that would be safer?